pkg/generate/app/app.go

```diff
@@ -109,32 +111,46 @@ func (r *SourceRef) SuggestName() (string, bool) {
 	if len(r.Name) > 0 {
 		return r.Name, true
 	}
-	return nameFromGitURL(r.URL)
+	if r.URL != nil {
+		return nameFromGitURL(r.URL)
+	}
+	return "", false
 }
 
 // BuildSource returns an OpenShift BuildSource from the SourceRef
 func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
-	return &buildapi.BuildSource{
+	triggers := []buildapi.BuildTriggerPolicy{
+		{
+			Type: buildapi.GitHubWebHookBuildTriggerType,
+			GitHubWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
+		{
+			Type: buildapi.GenericWebHookBuildTriggerType,
+			GenericWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
```

Comment:
still think this should move to be added generically to all buildconfigs (this and all the other triggers we always add, in one place:  configchange,webhook,genericwebhook.  imagechangetrigger is probably a special case).

but i guess it doesn't have to be in this PR.